packages/server/graphql/types/GitLabIntegration.ts

```diff
@@ -31,18 +36,23 @@ const GitLabIntegration = new GraphQLObjectType<any, GQLContext>({
     sharedProviders: {
       description: 'The non-global providers shared with the team or organization',
       type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(IntegrationProviderOAuth2))),
-      resolve: async ({userId}, _args, {dataLoader}) => {
-        const teamMembers = await dataLoader.get('teamMembersByUserId').load(userId)
-        const teamIds = teamMembers.map(({teamId}) => teamId)
-        const orgIds = Array.from(new Set(teamMembers.map(({orgId}) => orgId)))
-
-        const orgTeams = (await dataLoader.get('teamsByOrgIds').loadMany(orgIds))
-          .flat()
-          .filter(isValid)
-        const orgTeamIds = orgTeams.map(({teamId}) => teamId)
+      resolve: async (
+        {teamId}: {teamId: string},
+        _args: unknown,
+        {authToken, dataLoader}: GQLContext
+      ) => {
+        const viewerId = getUserId(authToken)
+        const team = await dataLoader.get('teams').load(teamId)
```

Comment:
+1 you can use Georg's new API `loadNonNull` instead of `load` here because this GitLabIntegration object lives on a TeamMember object & you can't have a teamMember without a team, so we know team is guaranteed non-null